src/main/java/picard/analysis/CollectInsertSizeMetrics.java

```diff
@@ -158,23 +166,18 @@
         else  {
             file.write(OUTPUT);
 
-            final int rResult;
-            if(HISTOGRAM_WIDTH == null) {
-                rResult = RExecutor.executeFromClasspath(
-                    Histogram_R_SCRIPT,
-                    OUTPUT.getAbsolutePath(),
-                    Histogram_FILE.getAbsolutePath(),
-                    INPUT.getName());
-            } else {
-                rResult = RExecutor.executeFromClasspath(
-                    Histogram_R_SCRIPT,
-                    OUTPUT.getAbsolutePath(),
-                    Histogram_FILE.getAbsolutePath(),
-                    INPUT.getName(),
-                    String.valueOf(HISTOGRAM_WIDTH) ); //Histogram_WIDTH is passed because R automatically sets Histogram width to the last
-                                                         //bin that has data, which may be less than Histogram_WIDTH and confuse the user.
+            final List<String> plotArgs = new ArrayList<>();
+            Collections.addAll(plotArgs, OUTPUT.getAbsolutePath(), Histogram_FILE.getAbsolutePath(), INPUT.getName());
+
+            if (HISTOGRAM_WIDTH != null) {
+                plotArgs.add(String.valueOf(HISTOGRAM_WIDTH));
+            }
+            else if (MIN_HISTOGRAM_WIDTH != null) {
+                final int max = (int) file.getAllHistograms().stream().mapToDouble(Histogram::getMax).max().getAsDouble();
```

Comment:
In the absence of MIN_HISTOGRAM_WIDTH, this would make the largest insert-size that appears in the histogram the de-facto width, right?
 If so that is a departure from the current behavior that looks at the median and add a few deviations...

Reply:
A couple of things.  Firstly, this line is only triggered if a) `HISTOGRAM_WIDTH` is null and b) `MIN_HISTOGRAM_WIDTH` is not null.  This is akin to the logic in `if (HISTOGRAM_WIDTH == null)` previously, where no 4th argument is passed to limit the range of the plot.  So it's actually identical to the previous behaviour.

The way the limiting works (I had to look to remind myself when writing this PR) is that the data in the `Histogram` is actually trimmed using median + deviations.  So while the plot does fall back to using the full range of the data written to disk, the range of that data has already been truncated.

Reply:
OK. thanks for the detail.